beaker-vue/src/plugins/theme.ts

```diff
@@ -68,6 +68,14 @@ export const BeakerThemePlugin: Plugin = {
                 }
                 applyTheme()
             }, {once: true});
+
+            // This helps adding the theme to the app element for children elements
+            // but doesn't work on initial state TODO dont merge until fixed
+            const appElement = document.querySelector('[data-v-app]');
```

Comment:
This is useful and aligns a bit with what newer frameworks do when toggling between between dark/light mode:

It allows CSS rules to do things such as:

```
[data-theme="dark"] &:hover {
        background-color: var(--surface-a);
    }
```

Instead of relying on the in-js injected theme property to then conditionally add classes from the template code.
I ended up using it only once in this PR.

I think this is the place to add it- the idea is that when the app loads or changes theme the property is updated and the style rules can react to that (light/dark mode specific-overrides).